kasa/device_factory.py

```diff
@@ -1,123 +1,88 @@
 import logging
 import time
 from typing import Any, Dict, Optional, Tuple, Type
 
 from .aestransport import AesTransport
-from .credentials import Credentials
-from .device_type import DeviceType
-from .exceptions import UnsupportedDeviceException
+from .deviceconfig import DeviceConfig
+from .exceptions import SmartDeviceException, UnsupportedDeviceException
 from .iotprotocol import IotProtocol
-from .klaptransport import KlapTransport, TPlinkKlapTransportV2
-from .protocol import BaseTransport, TPLinkProtocol
+from .klaptransport import KlapTransport, KlapTransportV2
+from .protocol import (
+    BaseTransport,
+    TPLinkProtocol,
+    TPLinkSmartHomeProtocol,
+    _XorTransport,
+)
 from .smartbulb import SmartBulb
-from .smartdevice import SmartDevice, SmartDeviceException
+from .smartdevice import SmartDevice
 from .smartdimmer import SmartDimmer
 from .smartlightstrip import SmartLightStrip
 from .smartplug import SmartPlug
 from .smartprotocol import SmartProtocol
 from .smartstrip import SmartStrip
 from .tapo import TapoBulb, TapoPlug
 
-DEVICE_TYPE_TO_CLASS = {
-    DeviceType.Plug: SmartPlug,
-    DeviceType.Bulb: SmartBulb,
-    DeviceType.Strip: SmartStrip,
-    DeviceType.Dimmer: SmartDimmer,
-    DeviceType.LightStrip: SmartLightStrip,
-    DeviceType.TapoPlug: TapoPlug,
-    DeviceType.TapoBulb: TapoBulb,
+_LOGGER = logging.getLogger(__name__)
+
+GET_SYSINFO_QUERY = {
+    "system": {"get_sysinfo": None},
 }
 
-_LOGGER = logging.getLogger(__name__)
 
+async def connect(*, host: Optional[str] = None, config: DeviceConfig) -> "SmartDevice":
+    """Connect to a single device by the given connection parameters.
 
-async def connect(
-    host: str,
-    *,
-    port: Optional[int] = None,
-    timeout=5,
-    credentials: Optional[Credentials] = None,
-    device_type: Optional[DeviceType] = None,
-    protocol_class: Optional[Type[TPLinkProtocol]] = None,
-) -> "SmartDevice":
-    """Connect to a single device by the given IP address.
-
-    This method avoids the UDP based discovery process and
-    will connect directly to the device to query its type.
-
-    It is generally preferred to avoid :func:`discover_single()` and
-    use this function instead as it should perform better when
-    the WiFi network is congested or the device is not responding
-    to discovery requests.
-
-    The device type is discovered by querying the device.
-
-    :param host: Hostname of device to query
-    :param device_type: Device type to use for the device.
-        If not given, the device type is discovered by querying the device.
-        If the device type is already known, it is preferred to pass it
-        to avoid the extra query to the device to discover its type.
-    :param protocol_class: Optionally provide the protocol class
-            to use.
-    :rtype: SmartDevice
-    :return: Object for querying/controlling found device.
+    Do not use this function directly, use SmartDevice.Connect()
```

Comment:
```suggestion
    Do not use this function directly, use SmartDevice.connect()
```
Just a typo fix, but we should still keep a proper docstring here even if we would recommend using `SmartDevice.connect()` for creating instances, so please adjust that.

Reply:
Done